src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncer.java

```diff
@@ -1,892 +1,39 @@
- *
- * http://aws.amazon.com/asl/
- *
- * or in the "license" file accompanying this file. This file is distributed
- * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied. See the License for the specific language governing
- * permissions and limitations under the License.
- */
 package com.amazonaws.services.kinesis.clientlibrary.lib.worker;
 
-import java.io.Serializable;
-import java.math.BigInteger;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-import org.apache.commons.lang3.StringUtils;
-
 import com.amazonaws.services.kinesis.clientlibrary.exceptions.internal.KinesisClientLibIOException;
 import com.amazonaws.services.kinesis.clientlibrary.proxies.IKinesisProxy;
-import com.amazonaws.services.kinesis.clientlibrary.types.ExtendedSequenceNumber;
 import com.amazonaws.services.kinesis.leases.exceptions.DependencyException;
 import com.amazonaws.services.kinesis.leases.exceptions.InvalidStateException;
 import com.amazonaws.services.kinesis.leases.exceptions.ProvisionedThroughputException;
 import com.amazonaws.services.kinesis.leases.impl.KinesisClientLease;
 import com.amazonaws.services.kinesis.leases.interfaces.ILeaseManager;
-import com.amazonaws.services.kinesis.metrics.impl.MetricsHelper;
-import com.amazonaws.services.kinesis.metrics.interfaces.MetricsLevel;
-import com.amazonaws.services.kinesis.model.Shard;
 
 /**
- * Helper class to sync leases with shards of the Kinesis stream.
- * It will create new leases/activities when it discovers new Kinesis shards (bootstrap/resharding).
- * It deletes leases for shards that have been trimmed from Kinesis, or if we've completed processing it
- * and begun processing it's child shards.
+ * Interface to sync leases with shards of the Kinesis stream.
  */
-class ShardSyncer {
-
-    private static final Log LOG = LogFactory.getLog(ShardSyncer.class);
-
-    /**
-     * Note constructor is private: We use static synchronized methods - this is a utility class.
-     */
-    private ShardSyncer() {
-    }
-
-    static synchronized void bootstrapShardLeases(IKinesisProxy kinesisProxy,
-            ILeaseManager<KinesisClientLease> leaseManager,
-            InitialPositionInStreamExtended initialPositionInStream,
-            boolean cleanupLeasesOfCompletedShards,
-            boolean ignoreUnexpectedChildShards)
-        throws DependencyException, InvalidStateException, ProvisionedThroughputException, KinesisClientLibIOException {
-        syncShardLeases(kinesisProxy, leaseManager, initialPositionInStream, cleanupLeasesOfCompletedShards,
-                        ignoreUnexpectedChildShards);
-    }
+public interface ShardSyncer {
```

Comment:
Doesn't need to be changed to an interface unless implementing a different `ShardSyncer`. The only reason I could see for `KinesisShardSyncer` to be created is because this class has a private method. If that is the only reason, I would recommend changing the constructor on the `ShardSyncer` rather than changing it to an interface.